packages/medusa/src/strategies/batch-jobs/product/import.ts

```diff
@@ -255,6 +268,36 @@ class ProductImportStrategy extends AbstractBatchJobStrategy {
     })
   }
 
+  /**
+   * Create, or retrieve by name, sales channels from the input data.
+   *
+   * @param data an array of sales channels partials
+   * @return an array of sales channels created or retrieved by name
+   */
+  private async processSalesChannels(
+    data: Pick<SalesChannel, "name" | "description">[]
+  ): Promise<SalesChannel[]> {
+    const transactionManager = this.transactionManager_ ?? this.manager_
+    const salesChannelServiceTx =
+      this.salesChannelService_.withTransaction(transactionManager)
+
+    const salesChannels: SalesChannel[] = []
+
+    for (const input of data) {
+      let sc = await salesChannelServiceTx.retrieveByName(input.name)
+
+      if (!sc) {
+        sc = await salesChannelServiceTx.create(
+          input as CreateSalesChannelInput
+        )
+      }
```

Comment:
thought: I don't think we should allow sales channel creation in the product import

Reply:
in that case, do we want only "sales channel id" column, or do we keep the import schema same as export (which has name and description)

Reply:
I like the approach of using the sales channel name and looking up based on that 👍

Reply:
We could even support both options:

|Sales Channel Name |Sales Channel ID | |
|---------------------|------------------|-|
|My Sales Channel     |  | ✅ This works if there is a sales channel named my sales channel |
| | sc_131231231233 |  ✅ This works if there is a sales channel with the id |